trigger/coap/trigger.go

```diff
@@ -0,0 +1,227 @@
+package coap
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/dustin/go-coap"
+	"github.com/project-flogo/core/data/metadata"
+	"github.com/project-flogo/core/support/log"
+	"github.com/project-flogo/core/trigger"
+)
+
+var triggerMd = trigger.NewMetadata(&Settings{}, &HandlerSettings{}, &Output{})
+
+func init() {
+	_ = trigger.Register(&CoApTrigger{}, &Factory{})
+}
+
+type Factory struct {
+}
+
+// Metadata implements trigger.Factory.Metadata
+func (*Factory) Metadata() *trigger.Metadata {
+	return triggerMd
+}
+
+// New implements trigger.Factory.New
+func (*Factory) New(config *trigger.Config) (trigger.Trigger, error) {
+	s := &Settings{}
+	err := metadata.MapToStruct(config.Settings, s, true)
+	if err != nil {
+		return nil, err
+	}
+
+	return &CoApTrigger{id: config.Id, settings: s}, nil
+}
+
+// Trigger CoAp trigger struct
+type CoApTrigger struct {
+	server    *Server
+	settings  *Settings
+	resources map[string]*CoapResource
+	id        string
+	logger    log.Logger
+}
+type CoapResource struct {
+	path     string
+	attrs    map[string]string
+	handlers map[coap.COAPCode]trigger.Handler
+}
+
+func (t *CoApTrigger) Initialize(ctx trigger.InitContext) error {
+
+	t.logger = ctx.Logger()
+
+	mux := coap.NewServeMux()
+
+	mux.Handle("/.well-known/core", coap.FuncHandler(t.handleDiscovery))
+
+	// Init handlers
+	for _, handler := range ctx.GetHandlers() {
+
+		s := &HandlerSettings{}
+		err := metadata.MapToStruct(handler.Settings(), s, true)
+		if err != nil {
+			return err
+		}
+
+		path := s.Path
+		method := s.Method
+
+		t.logger.Debugf("Registering handler [%s]", path)
+		resource, exists := t.resources[path]
+
+		if !exists {
+			resource = &CoapResource{path: path, attrs: make(map[string]string), handlers: make(map[coap.COAPCode]trigger.Handler)}
+			t.resources[path] = resource
+		}
+
+		resource.handlers[toCoapCode(method)] = handler
+
+		mux.Handle(path, newActionHandler(t, resource))
```

Comment:
this will overwrite the mux entry, you have to add the resources/actionhandlers to the kux in a seperate loop.